### PR TITLE
Feature/add approver

### PR DESCRIPTION
**Pull request by nogahaskal**
This PR was tested local and in here: http://noga.northeurope.cloudapp.azure.com.
New env: APPROVAL_URL -> this is the approval service url...
I added a request GetApproverInfo that gets the user information from the approval service. Then we can check if the user can approve his own requests. More then that we can check who can approve the user requests in his unit.
This is not breaking change! just more route...

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -183,10 +183,6 @@ services:
     depends_on:
       - mongo
       - redis
```
Comment (yosef32) [inline docker-compose.yml:185]: it depends on redis and you delete the redis?

File: user/user.go [hunk 2]
```diff
@@ -129,6 +130,34 @@ func (r *Router) SearchByName(c *gin.Context) {
...
+// GetApproverInfo is the request handler for GET /users/:id/approverInfo
+func (r *Router) GetApproverInfo(c *gin.Context) {
+	reqUser := ExtractRequestUser(c)
+	if reqUser == nil {
+		c.AbortWithStatus(http.StatusUnauthorized)
+		return
+	}
+	userID := c.Param(ParamUserID)
+	if userID == "" {
+		c.String(http.StatusBadRequest, "id is required")
```
Comment (yosef32) [inline user/user.go:142]: Better to use for the message fmt.Sprintf("%s is required", ParamUserID),
the ParamUserID value replace with %s

File: user/user.go [hunk 3]
```diff
@@ -129,6 +130,34 @@ func (r *Router) SearchByName(c *gin.Context) {
...
+		return
+	}
+
+	getApproverInfoRequest := &uspb.GetApproverInfoRequest{
+		Id: userID,
+	}
+
+	info, err := r.userClient.GetApproverInfo(c.Request.Context(), getApproverInfoRequest)
```
Comment (yosef32) [inline user/user.go:150]: in the proto file you create message ApproverInfo and you only use it once in GetApproverInfoResponse I think it's unnecessary, 
Just put the values of ApproverInfo into GetApproverInfoResponse

File: docker-compose.yml (hunk 1 above)
Comment (nogahaskal) [reply docker-compose.yml:185]: this key was duplycate

File: user/user.go (hunk 3 above)
Comment (nogahaskal) [reply user/user.go:150]: done.

File: user/user.go (hunk 2 above)
Comment (nogahaskal) [reply user/user.go]: done.

File: user/user.go [hunk 4]
```diff
@@ -129,6 +130,34 @@ func (r *Router) SearchByName(c *gin.Context) {
...
+// GetApproverInfo is the request handler for GET /users/:id/approverInfo
+func (r *Router) GetApproverInfo(c *gin.Context) {
+	reqUser := ExtractRequestUser(c)
+	if reqUser == nil {
+		c.AbortWithStatus(http.StatusUnauthorized)
+		return
+	}
+	userID := c.Param(ParamUserID)
+	if userID == "" {
+		c.String(http.StatusBadRequest, fmt.Sprintf("%s is required", ParamUserID))
```
Comment (Shahar-Y) [inline user/user.go:142]:
```suggestion
		c.String(http.StatusBadRequest, fmt.Sprintf("%s field is required", ParamUserID))
```

**Comment by Shahar-Y** [comment]
Add in the description what the PR does, was it tested and how, and is it affected by other PRs that could break the code

File: user/user.go (hunk 4 above)
Comment (nogahaskal) [reply user/user.go]: done :+1:

**Pull request merged**